server/app/views/applicant/ApplicantLayout.java

```diff
@@ -197,12 +197,26 @@ private NavTag renderNavBar(Http.Request request, Optional<String> userName, Mes
 
     String displayUserName = ApplicantUtils.getApplicantName(userName, messages);
     return nav()
-        .withClasses("bg-white", "border-b", "align-middle", "p-1", "grid", "grid-cols-3")
-        .with(branding())
-        .with(maybeRenderTiButton(profile, displayUserName))
+        .withClasses("bg-white", "border-b", "align-middle", "p-1", "flex", "flex-row", "flex-wrap")
         .with(
-            div(getLanguageForm(request, profile, messages), authDisplaySection(userName, messages))
-                .withClasses("justify-self-end", "flex", "flex-row"));
+            div()
+                .with(branding())
+                .withClasses(
+                    "items-center", "place-items-center", "flex-shrink-0", "grow", "md:grow-0"))
+        .with(maybeRenderTiButton(profile, displayUserName).withClasses("grow-0", "md:grow"))
+        .with(
+            div()
+                .with(
+                    getLanguageForm(request, profile, messages),
+                    authDisplaySection(userName, messages))
+                .withClasses(
+                    "flex",
+                    "flex-row",
+                    "grow",
+                    "md:grow-0",
+                    "shrink-0",
+                    "md:shrink",
+                    "place-content-center"));
```

Comment:
**suggestion:** Do we want to use a responsive breakpoint so the switch happens at a controlled time? Like https://github.com/civiform/civiform/blob/605f99320654d4c0f1348f359ed4e2d52dad6b37/server/app/views/applicant/ApplicantUpsellView.java#L94

Or just when there's room?

Reply:
Sure, added a few of them as appropriate